lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -132,25 +131,71 @@ export class BrowserCacheManager extends CacheManager {
      * This is used to migrate users from older versions of MSAL which did not create the map.
      * @returns 
      */
-    private createAccountKeyMap(): void {
+    private createKeyMaps(): void {
+        this.logger.trace("BrowserCacheManager - createKeyMaps called.");
         const accountKeys = this.getItem(StaticCacheKeys.ACCOUNT_KEYS);
-        if (accountKeys) {
-            // Account key map already exists, no need to iterate through cache
+        const tokenKeys = this.getItem(`${StaticCacheKeys.TOKEN_KEYS}.${this.clientId}`);
+        if (accountKeys && tokenKeys) {
+            this.logger.verbose("BrowserCacheManager:createKeyMaps - account and token key maps already exist, skipping migration.");
+            // Key maps already exist, no need to iterate through cache
             return;
         }
 
         const allKeys = this.browserStorage.getKeys();
         allKeys.forEach((key) => {
-            if (this.isAccountKey(key)) {
+            if (this.isCredentialKey(key)) {
+                // Get item, parse, validate and write key to map
                 const value = this.getItem(key);
                 if (value) {
-                    try {
-                        const accountObj = JSON.parse(value);
-                        if (AccountEntity.isAccountEntity(accountObj)) {
-                            this.addAccountKeyToMap(key);
+                    const credObj = this.validateAndParseJson(value);
+                    if (credObj && credObj.hasOwnProperty("credentialType")) {
+                        switch (credObj["credentialType"]) {
+                            case CredentialType.ID_TOKEN:
+                                if (IdTokenEntity.isIdTokenEntity(credObj)) {
```

Comment:
This means that there is a cache item that has "idtoken" in its key but not a valid one. Can we log a warning? Same applies below for other cred types.

Reply:
Logging added